src/inventorymanager.cpp

```diff
@@ -154,94 +170,257 @@ IMoveAction::IMoveAction(std::istream &is, bool somewhere) :
 	}
 }
 
-void IMoveAction::swapDirections()
-{
-	std::swap(from_inv, to_inv);
-	std::swap(from_list, to_list);
-	std::swap(from_i, to_i);
-}
-
-void IMoveAction::onPutAndOnTake(const ItemStack &src_item, ServerActiveObject *player) const
+static void onPutAndOnTake(const MoveAction& ma, const ItemStack &item, ServerActiveObject *player)
```

Comment:
What's the goal of making these functions static? They clearly belong to the `MoveAction` class.

Reply:
I don't remember XD.
Moved to class functions again.